packages/shared/lib/transactions.ts

```diff
@@ -132,6 +131,9 @@ export const isMempoolTx = (
   transaction: explorer.Transaction | explorer.MempoolTransaction
 ): transaction is explorer.MempoolTransaction => !('blockHash' in transaction)
 
+export const isInternalTx = (tx: explorer.Transaction, internalAddresses: AddressHash[]): boolean =>
+  [...(tx.outputs ?? []), ...(tx.inputs ?? [])].every((io) => io?.address && internalAddresses.indexOf(io.address) >= 0)
```

Comment:
Nit: I know it was already done like this before, but what about using `includes` instead of `indexOf` for readability sake?